Introduce webpack.config.js

joeheyming opened the pull request:
Webpack's job is to build a bundle.
Anywhere I saw am.blah = function() {}
I refactored the code to export the function or object with module.exports.
Then, I created index.js.

This exposes the am object.
Every module that set am.blah, now sets am.blah inside index.js.

Any function function that needed some external module,
now we require the external module at the top of the file.

Ran jshint, fixed a few issues.
Ran every demo, and reduced the <head> tag to only have
ani-machine.min.js

Added animate.css and cssshake as npm dependencies.
Then inside index.js, we require each of the css files.
The css loader allows us to import css inside javascript rather
than having to minify the css using gulp.

Deleted the gulpfile.

Made sure the webpack config exports am on the window object.
And also has a banner at the top.

Added classie as a npm dependency and require it from node_modules
instead of from the js libs folder

To run webpack build run:
npm run build

To watch it, run:
npm run watch.

To run jshint:

npm run jshint

Commit:
Introduce webpack.config.js

Webpack's job is to build a bundle.
Anywhere I saw am.blah = function() {}
I refactored the code to export the function or object with module.exports.
Then, I created index.js.

This exposes the am object.
Every module that set am.blah, now sets am.blah inside index.js.

Any function function that needed some external module,
now we require the external module at the top of the file.

Ran jshint, fixed a few issues.
Ran every demo, and reduced the <head> tag to only have
ani-machine.min.js

Added animate.css and cssshake as npm dependencies.
Then inside index.js, we require each of the css files.
The css loader allows us to import css inside javascript rather
than having to minify the css using gulp.

Deleted the gulpfile.

Made sure the webpack config exports am on the window object.
And also has a banner at the top.

Added classie as a npm dependency and require it from node_modules
instead of from the js libs folder

libs/events.js

jfroffice commented on js/index.js:
```diff
@@ -0,0 +1,23 @@
+};
+
+require('css/light.css');
+require('csshake/dist/csshake.min.css');
```
this dep should stay outside the library

jfroffice reviewed (commented):
Can you change `npm run watch` to `npm run dev`?

LGTM, except for the CSS shake library that should stay outside from this library 😀

Commit:
Remove csshake; update package.json.

csshake is supposed to stay outside the library.
So I removed it.
I rebuilt the js file.
Then I updated npm run watch to npm run dev.

joeheyming commented:
Done
